core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -320,6 +320,20 @@ object ZIOSpec extends ZIOBaseSpec {
         }
       }
     ),
+    suite("foldRight")(
```

Comment:
Can we add a couple more tests here for the ordering of the fold? Right now these tests would pass even if we were folding in the wrong order.